gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/KubernetesPluginTest.java

```diff
@@ -90,4 +92,15 @@ void getTaskPrecedence_withValidProject_shouldReturnTaskPrecedence() {
         .containsEntry("k8sHelmPush", Collections.singletonList(KubernetesHelmTask.class))
         .containsEntry("k8sHelmLint", Collections.singletonList(KubernetesHelmTask.class));
   }
+
+  @Test
```

Comment:
Hello,
Thanks for the contribution !
I don't understand what is the purpose of these tests. Are these verifying that the method that is called before is called ? Could you clarify ? thanks

Reply:
We have updated the method signature as part of current changes, and we need to make sure that the **register** method is correctly registering the tasks, even though now it won't return the task. ( return type changed to **void**)

In these tests, we have made sure when the **register** method is called on any AbstractJKubePlugin implementations ( KubernetesPlugin OR OpenShiftPlugin), it does not return anything and still register the task correctly.

Reply:
Sun is referring to mocking and verifying the object being tested (which looks quite odd). You shouldn't be mocking the object you're testing.

Reply:
Yes, my bad. Thanks for pointing it out. I have updated the test cases, please have a look once.

Reply:
sorry I don't understand
```
    jkubePlugin.register(project, "k8sApply", KubernetesApplyTask.class);
    verify(jkubePlugin, times(1)).register(project,"k8sApply", KubernetesApplyTask.class);
```
So this is verifying that `jkubePlugin.register` is called, but it is just done 1 line above. I don't see the point. To me you should remove these tests.

Reply:
I still think this is over testing. IMHO these tests are not required.

Reply:
I can still see modifications in test files (import and new line added)

Reply:
Fixed

Reply:
Thanks !!!